code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -950,11 +952,19 @@ Pass a positive integer as an argument to override a bot's default speed.
 /mob/living/simple_animal/bot/Login()
 	. = ..()
 	access_card.access += player_access
+
+	data_hud = GLOB.huds[data_hud_type]
+	data_hud.add_hud_to(src)
+
 	diag_hud_set_botmode()
 	show_laws()
 
 /mob/living/simple_animal/bot/Logout()
 	. = ..()
+
+	if(data_hud)
+		data_hud.remove_hud_from(src)
```

Comment:
Does this null out `data_hud` if the bot is destroyed/deleted while a player is in control of it? i.e doesn't cause a GC issue that `data_hud` is still assigned.

Reply:
@SteelSlayer  That's a good question. I don't think it will null it out implicitly so I can add an explicit statement here, would that make sense?

Reply:
I would add that in bot Destroy.

Reply:
Done, PTAL. I know that
```
if(data_hud)
	data_hud.remove_hud_from(src)
	QDEL_NULL(data_hud)
```
is expanded to 
```
if(data_hud)
	data_hud.remove_hud_from(src)
	if(data_hud) { qdel(data_hud); data_hud = null }
```
and thus has two `if(data_hud)` conditionals but it does avoid an extra line of code here. Dunno what the preferred style is here.

Reply:
I think what you have is fine, since bots being destroyed does not happen a whole lot.

Reply:
👍